### PR TITLE
refactor: harmonise loaders

### PR DESCRIPTION
**Pull request by santimirandarp**
The parsers were rewritten with this structure (metacode):
```
anyParser(filecollection, logger?:boolean){
 measurements = getEmptyMeasurements()
  logs = [ ] 
  for(loop over files){
   try{
     // try to parse
   } catch(e){
    // push to logs
   }
  return measurements //with the new ones
}
```

The code changes some previous additions from [4ff5580](https://github.com/zakodium-oss/analysis-ui-components/commit/4ff5580a5818a83bd88c078989a7397c45ee577e), just `assert` for an error log and Partial for full `Measurement` object, that's why I included the template, in case it needs discussion.

### CODE REVIEW

File: src/app-data/DataState.ts [hunk 1]
```diff
@@ -63,11 +63,12 @@ export const kindsLabel: Record<MeasurementKind, string> = {
...
   fileCollection: FileCollection,
-) => Promise<Partial<Measurements>>;
+  logger?: boolean,
+) => Promise<Measurements>;
 
 export function mergeMeasurements(
-  measurements: Partial<Measurements>,
-  newMeasurements: Partial<Measurements>,
+  measurements: Measurements,
+  newMeasurements: Measurements,
```
Comment (targos) [inline src/app-data/DataState.ts:71]: Please don't revert this improvement.

**Comment by targos** [comment]
It looks like you reverted the changes from https://github.com/zakodium-oss/analysis-ui-components/commit/4ff5580a5818a83bd88c078989a7397c45ee577e.

File: src/app-data/DataState.ts (hunk 1 above)
Comment (santimirandarp) [reply src/app-data/DataState.ts:71]: If all the loaders return `Measurements` wouldn't it need to be removed? Previously there were some returning Partial Measurements, and some the full object.

**Comment by santimirandarp** [comment]
Not sure whether we could change this

> Error: src/app-data/appState.tsx(103,50): error TS2345: Argument of type 'Partial<Measurements>' is not assignable to parameter of type 'Measurements'.

To accept Measurements, which seems the error. 

> It looks like you reverted the changes from [4ff5580](https://github.com/zakodium-oss/analysis-ui-components/commit/4ff5580a5818a83bd88c078989a7397c45ee577e).

I think some of them yes, that I why I included the template, in case it needs discussion, basically I put all of them to return the same object as that seems easier from my point of view at least.

File: src/app-data/DataState.ts (hunk 1 above)
Comment (targos) [reply src/app-data/DataState.ts:71]: All loaders should return `Partial<Measurements>`.

Comment (santimirandarp) [reply src/app-data/DataState.ts:71]: I see. Basically all those changes were triggered by assuming that `const newMeasurements =  getEmptyMeasurement()` was correct and would help making all the loaders the same, in which case the return would be Measurements. I will try to understand better what it was intended.

**Pull request closed without merging**